wodles/azure/tests/test_orm.py

```diff
@@ -0,0 +1,192 @@
+            for md5 in last_dates_dict[key].keys():
+                assert isinstance(last_dates_dict[key][md5], dict)
+                assert set(last_dates_dict[key][md5].keys()) == {'min', 'max'}
+
+
+@patch('os.path.exists', return_value=False)
+@patch('builtins.open')
+@patch('json.dump')
+def test_load_dates_json_no_file(mock_dump, mock_open, mock_exists):
+    orm.load_dates_json()
+    mock_exists.assert_called_once()
+    mock_open.assert_called_once()
+    mock_dump.assert_called_once_with(orm.last_dates_default_contents, mock_open().__enter__())
+
+
+@pytest.mark.parametrize('last_dates_file_path', [
+    (os.path.join(test_last_dates_path, 'last_dates_invalid.json'))
+])
+def test_load_dates_json_ko(last_dates_file_path):
+    with patch('orm.last_dates_path', new=last_dates_file_path):
+        with pytest.raises(json.JSONDecodeError):
+            orm.load_dates_json()
+
+
+@pytest.mark.parametrize('file_exists, file_size', [
+    (True, 0),
+    (True, 100),
+    (False, 100),
+])
+@patch('orm.create_db')
+@patch('orm.migrate_from_last_dates_file')
+def test_check_integrity(mock_migrate, mock_create_db, create_and_teardown_db, file_exists, file_size):
+    with patch('os.path.exists', return_value=file_exists):
+        with patch('os.path.getsize', return_value=file_size):
+            orm.check_database_integrity()
+            mock_create_db.assert_called()
+            if file_exists and file_size > 0:
+                mock_migrate.assert_called()
+            else:
+                mock_migrate.assert_not_called()
+
+
+def test_check_integrity_ko(teardown_db):
+    with patch('os.path.exists'):
+        with patch('os.path.getsize', return_value=100):
+            with patch('orm.migrate_from_last_dates_file', side_effect=Exception):
+                assert orm.check_database_integrity() is False
+
+
+@pytest.mark.parametrize('last_dates_file_path', [
+    (os.path.join(test_last_dates_path, 'last_dates.json')),
+    (os.path.join(test_last_dates_path, 'last_dates_graph.json')),
+    (os.path.join(test_last_dates_path, 'last_dates_log_analytics.json')),
+    (os.path.join(test_last_dates_path, 'last_dates_storage.json')),
+    (os.path.join(test_last_dates_path, 'last_dates_old.json')),
+    (os.path.join(test_last_dates_path, 'last_dates_clean.json'))
+])
+def test_migrate_from_last_dates_file(last_dates_file_path, create_and_teardown_db):
+    """Test the last_dates file migration functionality."""
+    items = orm.get_all_rows(table=orm.Graph)
+    assert len(items) == 0
+
+    with patch('orm.last_dates_path', new=last_dates_file_path):
+        test_file_contents = orm.load_dates_json()
+        orm.migrate_from_last_dates_file()
+
+    # Validate the contents of each table
+    for table in [orm.Graph, orm.LogAnalytics, orm.Storage]:
+        items = orm.get_all_rows(table=table)
+        if table.__tablename__ in test_file_contents:
+            assert len(items) == len(test_file_contents[table.__tablename__].keys())
```

Comment:
You are not testing that the rows inserted have the expected content, which I think is the point of this test.

Reply:
You are right. That was missing. Done